test/test_jit.py

```diff
@@ -1796,6 +1796,68 @@ def forward(self, x):
         self.assertEqual(out, out_state)
         self.assertNotEqual(out, out_ones)
 
+    def test_shape_prop_mismatch_output(self):
+        with self.assertRaisesRegex(RuntimeError, "Number of op outputs did not match number of node outputs"):
+            cu = torch.jit.CompilationUnit('''
+            def test_shape_prop_mismatch_output(a) -> (b):
+                b = slice(a, dim=0, end=-2, start=2, step=1)
+                b = topk(a, dim=0, k=2, largest=True, sorted=True)
+            ''')
+            inputs = [torch.zeros(10)]
+            outputs = [torch.zeros(2), torch.from_numpy(np.array([1, 5])).long()]
+
+            real_outs = cu.test_shape_prop_mismatch_output(*inputs)
+            self.assertEqual(real_outs, outputs)
+
+    def test_view_shape_prop(self):
+        cu = torch.jit.CompilationUnit('''
+        def test_view_shape_prop(a) -> (b):
+            b = view(a, size=[-1])
+        ''')
+        inputs = [torch.zeros(10, 10)]
+        outputs = torch.zeros(100)
+
+        real_outs = cu.test_view_shape_prop(*inputs)
+        self.assertEqual(real_outs, outputs)
+
+    def test_integral_shape_inference(self):
+        cu = torch.jit.CompilationUnit('''
+        def test_integral_shape_inference(a) -> (b):
+            b = a / a
+        ''')
+        inputs = [torch.ones(10, 10).type(torch.LongTensor)]
+        outputs = torch.ones(10, 10)
+
+        self.assertEqual(cu.test_integral_shape_inference(*inputs), outputs)
+
+    def test_fuser_multiple_blocks(self):
+        cu = torch.jit.CompilationUnit('''
+        def test_fuser_multiple_blocks(this, that, theother, meme) -> (this, that, theother):
+            i = 0LL
+            while i < 20LL:
+                this = cat(this, meme, dim=0)
+                that = cat(that, meme, dim=0)
+                theother = cat(theother, meme, dim=0)
+                print(i)
+                i = i + 1
+        ''')
+
+        inputs = [torch.ones(0, 10, 10)] * 3
+        inputs += [torch.ones(1, 10, 10)]
+        outputs = [torch.ones(20, 10, 10)] * 3
+
+        self.assertEqual(cu.test_fuser_multiple_blocks(*inputs), outputs)
+
+    def test_print_graph_executor(self):
+        cu = torch.jit.CompilationUnit('''
```

Comment:
This should capture stdout and check that it printed what we expected. Code for capturing stdout is at the top of this file.